sdk/core/azure_core_amqp/Cargo.toml

```diff
@@ -49,6 +49,8 @@ fe2o3_amqp = [
   "serde_amqp",
   "serde_bytes",
 ]
+native_tls = ["fe2o3-amqp?/native-tls"]
```

Comment:
It doesn't seem there's a choice, but any `native-tls` feature would have to be sans provider like we do in `typespec_client_core`'s `reqwest_rustls` feature e.g., `reqwest/rustls-tls-native-roots-no-provider`. We absolutely cannot take a dependency on `ring`.

Now, it could be it's not actually used, but this would need to be investigated further.